envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/listeners/filters/RBACFilterFactoryTest.kt

```diff
@@ -412,6 +429,54 @@ internal class RBACFilterFactoryTest {
         assertThat(generated).isEqualTo(expectedRbacBuilder)
     }
 
+    @Test
+    fun `should generate RBAC rules for incoming permissions with source ip and selector authentication`() {
+        // given
+        val expectedRbacBuilder = getRBACFilter(expectedSourceIpWithSelectorAuthPermissionsJson)
+        val incomingPermission = Incoming(
+                permissionsEnabled = true,
+                endpoints = listOf(IncomingEndpoint(
+                        "/example",
+                        PathMatchingType.PATH,
+                        setOf("GET"),
+                        setOf("client2:selector")
+                ))
+        )
+
+        // when
+        val generated = rbacFilterFactoryWithSourceIpWithSelectorAuth.createHttpFilter(
+                createGroup(incomingPermission),
+                snapshotForSourceIpAuth
+        )
+
+        // then
+        assertThat(generated).isEqualTo(expectedRbacBuilder)
+    }
+
+    @Test
+    fun `should generate RBAC rules for incoming permissions with source ip from discovery and selector authentication`() {
+        // given
+        val expectedRbacBuilder = getRBACFilter(expectedSourceIpFromDiscoveryWithSelectorAuthPermissionsJson)
```

Comment:
Maybe it would be better if we merge the two tests? It will test the same conditions, but it will additionally test combining multiple types of authentication

Reply:
It is a unit test, when a unit test fails it should point directly to the bug "area". Combining functionalities in unit tests will result in all green / all red tests. We should avoid that.

Reply:
ok, but are we sure that combining multiple types works then? Or if multiple ip ranges for given client work? I just found a potential bug which will only appear if we have multiple ipRanges for a client...

Reply:
I've created an additional case that combines both ip from range and discovery.

Reply:
thanks